contractcourt/channel_arbitrator.go

```diff
@@ -1194,10 +1228,13 @@ func (c *ChannelArbitrator) checkCommitChainActions(height uint32,
 			continue
 		}
 
-		toChain := c.shouldGoOnChain(
+		toChain, err := c.shouldGoOnChain(
 			htlc.RefundTimeout, c.cfg.IncomingBroadcastDelta,
-			height,
+			height, htlc.RHash,
```

Comment:
Could it be worth to split `shouldGoOnChain` in two versions, one for incoming and one for outgoing? Then we also don't need to pass in the hash and do the payment lookup.

Reply:
> Could it be worth to split `shouldGoOnChain` in two versions

I will give it a try and see if it is more clear this way.


> Then we also don't need to pass in the hash and do the payment lookup

We will need the lookup anyway to differentiate forward from initiated payment.

Reply:
Isn't an incoming htlc never an initiated payment?

Reply:
> Isn't an incoming htlc never an initiated payment?

Indeed, it wasn't clear to me that you intended to avoid the lookup only for the incoming. Understood now.

Reply:
I passed the htlc struct to the `shouldGoToChain` method so we can take a decision there separately for incoming/outgoing htlcs and avoid the query in case of incoming.